### DISCUSSION TITLE
Possible issue? Tensorboard must be separately installed

### DISCUSSION THREAD

**Discussion by RCF42** [MPoL-dev/MPoL#56] [Q&A] [answered]
Not sure if this is just me (which is why I put it here), but with the crossvalidation tutorial the line:

```
from torch.utils.tensorboard import SummaryWriter

```
returns an error saying there is no module called tensorboard and so you have to separately install it. Nothing major, really just want to see if anyone else is getting this (I tried it under a different directory and a fresh install of MPoL (from git not PyPl)

**Comment by iancze** [comment, marked as answer]
This is a good question and something we should clarify in the tutorial. MPoL has only a few core dependencies, like Pytorch and numpy. In general, I think this is preferred to allow as many people to install it without hassle as possible, and it helps stability in the long run.

But there are several optional dependencies that a user might want to install to visualize the output. For example, matplotlib, or tensorboard in this case. We probably should just make a note in the tutorial that this can be installed if the user wants.

**Reply by iancze** [reply]
If you were to install MPoL with .[dev] like in the developer docs, does this still happen?

**Reply by iancze** [reply]
And, in the specific example of the cross validation tutorial, we do provide an option to use the summary writer but don't actually use it or comment on it directly in that document. We should add this discussion as an improvement to the tutorial.

**Reply by RCF42** [reply]
No it doesn't happen for me with the .[dev] install. Thanks for the reply
